mysql-test/suite/innodb/t/percona_sys_tablespace_encrypt.test

```diff
@@ -222,6 +254,86 @@ SELECT @@innodb_sys_tablespace_encrypt;
 ALTER TABLE t9 TABLESPACE=`innodb_system`;
 SHOW CREATE TABLE t9;
 
+DROP TABLE t8,t9;
+
+--echo # Test the behaviour when innodb_file_per_table is disabled
+
+SET @save_innodb_file_per_table = @@global.innodb_file_per_table;
+SET GLOBAL innodb_file_per_table = 0;
+
+--echo # Setting innodb_file_per_table=0 forces table to be created using system tablespace unless
+--echo # explicilty tablespace=innodb_file_per_table is mentioned
+
+--echo # Must fail as it tries to create table in system tablespace which is a shared tablespace
+--error ER_ILLEGAL_HA_CREATE_OPTION
+CREATE TABLE t2 (a int) ENCRYPTION='Y';
+--error ER_ILLEGAL_HA_CREATE_OPTION
+CREATE TABLE t3 (a int) ENCRYPTION='N';
+
+--echo # Must be successful
+CREATE TABLE t2 (a int) TABLESPACE = innodb_file_per_table, ENCRYPTION='Y';
+CREATE TABLE t3 (a int) TABLESPACE = innodb_file_per_table, ENCRYPTION='N';
+DROP TABLE t2,t3;
+
+CREATE TABLE t2 (a int);
+--echo # Try to move the table from encrypted sys to un-encrypted file_per_table space 
+--echo # Must fail as the source is encrypted but the destination is not
+--error ER_TARGET_TS_UNENCRYPTED
+ALTER TABLE t2 TABLESPACE = innodb_file_per_table;
+
+--echo # Try to move the table from encrypted sys to encrypted file_per_table space
+--echo # Must be successful
+ALTER TABLE t2 TABLESPACE = innodb_file_per_table, ENCRYPTION='Y';
+DROP TABLE t2;
+
+--echo # Try to move the table from encrypted sys to un-encrypted external tablespace
+--echo # Must fail as the source is encrypted but the destination is not
+CREATE TABLESPACE tb01 ADD DATAFILE 'tb01.ibd' ENGINE=INNODB;
+CREATE TABLE t2 (a int);
+--error ER_TARGET_TS_UNENCRYPTED
+ALTER TABLE t2 TABLESPACE = tb01;
```

Comment:
can you add another variant with explicit ENCRYPTION='N'

Reply:
Added